### PR TITLE
Made a Plant game

### PR DESCRIPTION
**Pull request by ggao007**
**Issue:**
#8 

**Solution:**
Add a button that, when clicked, opens a modal with the plant game.

**Test Method:** 
Manual Test. Testing will be added in a follow-up PR.
 
**Risk:**
[Enter any potential risks identified here]

**Reviewed By:**
[Edit the message as to who reviewed it here]

**Additional Context**
If you want to run this code, just add the button like this in the app.js, I am waiting for the todolist page, I need to add this button into todolist page.
image

### CODE REVIEW

**Review by Denijar — COMMENTED** [review]
Great effort this is looking good. The file structure doesn't match our conventions, but we are going to have a Discord chat about that to fix it! Should easy, just a refactor and change the styling to use a module.scss file.

File: client/src/App.js [hunk 1]
```diff
@@ -1,7 +1,6 @@
 import React from "react";
 import logo from "./logo.svg";
 import "./style.css";
-
```
Comment (Denijar) [inline client/src/App.js:4]: To avoid changes, there should not be any unnecessary changes in the App.js file.

File: client/src/components/PlantTrees/stateUtils.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        "https://i.ibb.co/CnQgYZp/Tree2.png",
+        "https://i.ibb.co/2s4TDYc/Tree3.png",
+        "https://i.ibb.co/5r5tqWY/Tree4.png",
+    ],
+    buttonImageUrls: {
+        water: "https://i.ibb.co/C04zg8V/Watering-Can.png",
+    },
+    feedback: {
+        start:
+            "Dear, When you finish your task, you will get a Water Coin, you can use this coin to plant your own trees! Try to click the button!",
```
Comment (Denijar) [inline client/src/components/PlantTrees/stateUtils.js:16]: Let's talk about this wording during our Discord chat

File: client/src/components/PlantTrees/PlantButton/PlantTreesModal/PlantTrees.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import "./PlantTrees.css";
+
+export default function PlantTrees() {
+    const [config, setConfig] = useState({
+        plantSize: 0, //This var shows different trees pictures
+        points: 0, //This var control the progressbar
+        waterCoins: 20, // when the user complete one task they get some waterCoins, which can be used to water trees
+        feedback: stateUtils.feedback.start,
+    });
+    const POINTS = {
```
Comment (Denijar) [inline client/src/components/PlantTrees/PlantButton/PlantTreesModal/PlantTrees.js:15]: Nice commenting and use of constants.

**Review by Denijar — CHANGES_REQUESTED** [review]

**Review by LincRui — APPROVED** [review]

File: client/src/components/PlantButton/stateUtils.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+module.exports = {
+    controls: {
+        water: "water",
+    },
+    plantImageUrls: [
+        "https://i.ibb.co/DVP9kkq/Tree1.png",
+        "https://i.ibb.co/CnQgYZp/Tree2.png",
+        "https://i.ibb.co/2s4TDYc/Tree3.png",
+        "https://i.ibb.co/5r5tqWY/Tree4.png",
```
Comment (Denijar) [inline client/src/components/PlantButton/stateUtils.js:9]: Is there are reason why we don't include the images in the repository directly? Perhaps there should be an `images` folder inside the `Plant` folder that contains the images. I am worried that these i.ibb links might expire and then we will not be able to see the images.

File: client/package-lock 2.json [hunk 5]
```diff
@@ -0,0 +1,18818 @@
+{
```
Comment (Denijar) [inline client/package-lock 2.json:1]: Is this a second package-lock file? Can this be deleted?

File: client/src/components/PlantButton/stateUtils.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    ],
+    buttonImageUrls: {
+        water: "https://i.ibb.co/C04zg8V/Watering-Can.png",
+    },
+    feedback: {
+        start:
+            "Dear, When you finish your task, you will get a Water Coin, you can use this coin to plant your own trees! Try to click the button!",
+        win: "Congratulations! You are now a certified plant Mom.",
+        noCoin: "Sorry, You do not have enough coins to water the tree.",
```
Comment (Denijar) [inline client/src/components/PlantButton/stateUtils.js:18]: Better word: **Sorry, you need to complete more tasks to water the tree.**
I think that we should not mention the word "coin" because there is no information on the page that indicates the user has coins.

**Review by Denijar — CHANGES_REQUESTED** [review]

File: client/src/App.js (hunk 1 above)
Comment (Denijar) [inline client/src/App.js:4]: To avoid conflicts, there should be no changes in this file

**Review by Denijar — CHANGES_REQUESTED** [review]

File: client/src/App.js (hunk 1 above)
Comment (ggao007) [reply client/src/App.js:4]: Actually, I didn't add any code to App.js... if u find some conflicts, u can just delete..

**Review by Denijar — APPROVED** [review]
Amazing job with this :)
